Tests/NIOTests/SocketAddressTest.swift

```diff
@@ -24,6 +24,29 @@ class SocketAddressTest: XCTestCase {
         XCTAssertEqual("[IPv4]foobar.com:12345", sa.description)
     }
 
+    func testIn6AddrDescriptionWorks() throws {
+        let sampleString = "::1"
+        let sampleIn6Addr: [UInt8] = [ // ::1
+            0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
+            0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x70, 0x0, 0x0, 0x54,
+            0xc2, 0xb5, 0x58, 0xff, 0x7f, 0x0, 0x0, 0x7, 
+            0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x40, 0x1, 0x1, 0x0
+        ]
+
+        var address         = sockaddr_in6()
+        address.sin6_len    = UInt8(MemoryLayout<in6_addr>.size)
```

Comment:
`sin6_len` is not a POSIX `sockaddr` field, it's present only on macOS, so this compile will fail on Linux.

It may be easier to obtain a `sockaddr` structure like this:

```swift
guard case .v6(let addr) SocketAddress(ipAddress: "::1", port: 80) else { fatalError() }
let s = addr.address.addressDescription()
```

Reply:
> sin6_len is not a POSIX sockaddr field, it's present only on macOS

Yes, I know. And the attentive observer will also notice that it is filled incorrectly. 🤫 don't tell anyone (and if you have to, mention that it was late and that it was a 10 line test for a 1 line fix).

> It may be easier to obtain a sockaddr structure ... from a string

Since your string handling was b0rked in the first place, I intentionally avoided that. Stuff like that makes tests run through just because the bugs are balanced.

Anyways, I submitted yet another fix which also builds on Docker/macOS (the check should really be `#if Darwin` or something, but I guess we can live with that today and fix it tomorrow when porting to other platforms).
Hope it finally gets through the CI.